src/server/api/routes/users/v2/email.js

```diff
@@ -98,11 +99,23 @@ router.put('/:wallet/email', async (req, res, next) => {
       if (emailVerifyToken !== token) {
         throw new Error('INVALID_TOKEN');
       }
-      await t.update(userRef, {
-        email: emailUnconfirmed,
-        emailUnconfirmed: FieldValue.delete(),
-        emailVerifyToken: FieldValue.delete(),
+      const snapshot = await nftMintSubscriptionCollection
+        .where('subscriberEmail', '==', emailUnconfirmed)
+        .get();
+      const emailFollowees = snapshot.docs.map(doc => {
+        const { subscribedWallet } = doc.data();
+        return subscribedWallet;
       });
+      const promises = snapshot.docs.map(doc => t.delete(doc.ref));
+      promises.push(
+        t.update(userRef, {
+          email: emailUnconfirmed,
+          emailUnconfirmed: FieldValue.delete(),
+          emailVerifyToken: FieldValue.delete(),
+          followees: FieldValue.arrayUnion(...emailFollowees),
+        })
+      );
+      await Promise.all(promises);
```

Comment:
Note that only 500 write/update/delete ops can be done in a single transaction, we probably won't hit it here but maybe need some sanity check, or just use batched write instead of transaction to handle migration